float32 kp
float32 ki
float32 kd
float32 desired_mass